include/boost/geometry/algorithms/detail/is_valid/has_duplicates.hpp

```diff
@@ -48,7 +48,8 @@ struct has_duplicates
         geometry::equal_to<typename boost::range_value<Range>::type> equal;
 
         iterator it = boost::begin(view);
-        iterator next = ++boost::begin(view);
+        iterator next = it;
+        ++next;
```

Comment:
Similarly here, I would write like 50 and 51 as:
```
iterator it = boost::const_begin(view);
iterator next = ++boost::const_begin(view);
```
Also in the for-loop at line 52 I would use `next != boost::const_end(view)`.